x-pack/plugins/stack_connectors/server/connector_types/inference/inference.ts

```diff
@@ -83,29 +115,162 @@ export class InferenceConnector extends SubActionConnector<Config, Secrets> {
-      schema: ChatCompleteParamsSchema,
-    });
+  /**
+   * responsible for making a esClient inference method to perform chat completetion task endpoint and returning the service response data
+   * @param input the text on which you want to perform the inference task.
+   * @signal abort signal
+   */
+  public async performApiUnifiedCompletion(
+    params: UnifiedChatCompleteParams & { signal?: AbortSignal }
+  ): Promise<UnifiedChatCompleteResponse> {
+    const res = await this.performApiUnifiedCompletionStream(params);
+
+    const v = from(eventSourceStreamIntoObservable(res as Readable)).pipe(
+      filter((line) => !!line && line !== '[DONE]'),
+      map((line) => {
+        return JSON.parse(line) as OpenAI.ChatCompletionChunk | { error: { message: string } };
+      }),
+      tap((line) => {
+        if ('error' in line) {
+          throw new Error(line.error.message);
+        }
+        if (
+          'choices' in line &&
+          line.choices.length &&
+          line.choices[0].finish_reason === 'length'
+        ) {
+          throw new Error('createTokenLimitReachedError()');
+        }
+      }),
+      filter((line): line is OpenAI.ChatCompletionChunk => {
+        return 'object' in line && line.object === 'chat.completion.chunk';
+      }),
+      mergeMap((chunk): Observable<UnifiedChatCompleteResponse> => {
+        const events: UnifiedChatCompleteResponse[] = [];
+        events.push({
+          choices: chunk.choices.map((c) => ({
+            message: c.delta,
+            finish_reason: c.finish_reason,
+            index: c.index,
+          })),
+          id: chunk.id,
+          model: chunk.model,
+          object: chunk.object,
+          usage: chunk.usage,
+        } as UnifiedChatCompleteResponse);
+        return from(events);
+      }),
+      identity
+    );
+
+    const shared$ = v.pipe(share());
+
+    return lastValueFrom(
+      merge(
+        shared$,
+        shared$.pipe(
+          scan(
+            (prev, chunk) => {
+              if (chunk.choices.length > 0 && !chunk.usage) {
+                prev.choices[0].message.content += chunk.choices[0].message.content ?? '';
+
+                chunk.choices[0].message.tool_calls?.forEach((toolCall) => {
+                  let prevToolCall = prev.choices[0].message.tool_calls;
+                  if (!prevToolCall) {
+                    prev.choices[0].message.tool_calls = [
+                      {
+                        function: {
+                          name: '',
+                          arguments: '',
+                        },
+                        id: '',
+                      },
+                    ];
+
+                    prevToolCall = prev.choices[0].message.tool_calls;
+                  }
+
+                  prevToolCall.function.name += toolCall.function?.name;
+                  prevToolCall.function.arguments += toolCall.function?.arguments;
+                  prevToolCall.toolCall.id += toolCall.id;
```

Comment:
I think something is wrong here.

```
# prevToolCall 
[
  {
    "function": {
      "name": "",
      "arguments": ""
    },
    "id": ""
  }
]
# toolCall 
{
  "index": 0,
  "id": "call_aw16nHjsfySgIxC0WELXzmMD",
  "function": {
    "arguments": "",
    "name": "AlertCountsTool"
  },
  "type": "function"
}
```

`prevToolCall` is an array, but you are assigning properties to it. Maybe you mean to do `prevToolCall[0].function.name += ...`?

Reply:
```suggestion
                  prevToolCall[0].function.name += toolCall.function?.name;
                  prevToolCall[0].function.arguments += toolCall.function?.arguments;
                  prevToolCall[0].id += toolCall.id;
```
Not sure this is right but maybe?